### ISSUE TITLE
One hot encoded labels or logits?

### ISSUE THREAD

**Issue by kulkarnikeerti** [jkvt2/hierloss#3]
Hi, thanks for the loss implementation.
I am bit confused here. I know that labels are one hot encoded before we calculate the loss, I would also like to know if the logits are also in the form of one-hot encoding?
https://github.com/jkvt2/hierloss/blob/dc22dbbb4c33d8f271d985df779df8a8450b0f3c/hierarchical_loss.py#L27

PS: My final class prediction is equal to the number of classes, and I can use `cross_entropy` to calculate a loss in the normal case. If I use this hierarchical loss how are the predictions expected for the loss calculation? Is it a single value or a `len(num_classes)`?
